modules/doc/content/getting_started/installation/manual_installation_linux_lldb.md

```diff
@@ -0,0 +1,390 @@
+# Manual Installation for Linux LLDB
+
+!alert note
+These instructions have only been vetted on Ubuntu 19.10, but some of the
+concepts should be generally applicable. These
+instructions are really only intended for developers who cannot stand how slow
+`gdb` is on large libraries/executables, and prefer the speed of `lldb`. (Clang
+may also be faster at compiling templates, of which there are a fair amount in
+libMesh/MOOSE).
+
+## Compilers
+
+Install a clang compiler. You can choose to do this manually, but it's probably
+easier to just use your package manager. On Ubuntu based Linux systems:
+
+!package! code
+sudo apt install clang-9
+!package-end!
+
+Ubuntu generally has several options for clang compilers, for instance on 19.10
+there are clang-6, clang-7, clang-8, clang-9 and a default clang package. I
+elect to go with the newest available package. After installation:
+
+!package! code
+export CC=clang-9
+export CXX=clang++-9
+!package-end!
+
+Now in order to use various (optional) PETSc packages, we must also have an
+available fortran compiler. The de facto choice is `gfortran`:
+
+!package! code
+sudo apt install gfortran
+export FC=gfortran
+!package-end!
+
+## Compiler flags
+
+Now there are some subtleties here. At least on my system by default the clang
+compilers do not generate position-independent code or position-independent
+executables, e.g. by default clang is compiling as if with the
+`-fno-PIC -fno-PIE` flags. With these default settings, configuration of fortran
+support for MPICH will fail because of an inability to link objects generated by
+`gfortran` and objects generated by `clang`. To fix this we need to explicitly
+pass the `-fPIC -fPIE` flags:
+
+!package! code
+export CPPFLAGS='-fPIC -fPIE'
+!package-end!
+
+At the time of writing, `clang`
+[is known to have issues](https://bugzilla.redhat.com/show_bug.cgi?id=1740927)
+reading debug info from `libstdc++`. The most straightforward fix to this is to
+have `clang` institute its own debug information:
+
+!package! code
+export CFLAGS='-fstandalone-debug'
+export CXXFLAGS='-fstandalone-debug'
+!package-end!
+
+Note that we do not pass this option into `CPPFLAGS` because `gfortran` doesn't
```

Comment:
Us pass it into `CXXFLAGS` above

Reply:
Huh, yea that is intentional. We explicitly pass it to the C compiler with `CFLAGS` and to the CXX compiler with `CXXFLAGS`. But `CPPFLAGS` is used by the C, CXX, and fortran  compilers, so if we don't want the fortran compiler compiling with that flag, then we make sure its not in `CPPFLAGS`

Reply:
D'uh, "...do **not** pass..."